sphinx/writers/latex.py

```diff
@@ -1908,22 +1908,16 @@ def add_target(id):
                         node.parent.parent.index(node.parent)]
                 else:
                     raise
-            if isinstance(next, nodes.section):
+            domain = self.builder.env.get_domain('std')
+            figtype = domain.get_figtype(next)
+            if figtype and domain.get_numfig_title(next):
```

Comment:
This block is still needed only for captioned `literal_block`. It seems they are not broken. So I will refactor them on master branch later.

Reply:
see  #5008.